thin-replica-server/include/thin-replica-server/thin_replica_impl.hpp

```diff
@@ -274,6 +274,34 @@ class ThinReplicaImpl {
       return subscribe_status;
     }
 
+    // If last_known + 1 update requested keep waiting until we have at least one live update
+    // Note that TRS considers last_known + 1 update request as valid, even if the requested
+    // update doesn't exist in storage yet.
+    bool next_update_available = false;
```

Comment:
Suggestion: No need for the extra variable. As soon as we get an update we could `break` from the loop.